src/components/NoteForm/NoteForm.js

```diff
@@ -1,20 +1,46 @@
+import { useNote } from "../../contexts";
+import { getFormattedDate } from "../../utils";
 import "./NoteForm.css";
 
-const NoteForm = () => {
+const NoteForm = ({ note, setNote, addNoteHandler, updateNoteHandler }) => {
+  const {
+    noteState: { editedNotes },
+  } = useNote();
+
   return (
     <>
-      <form className="note-form-wrapper my-4">
+      <form
+        className="note-form-wrapper my-4"
+        onSubmit={
+          editedNotes
+            ? (e) => updateNoteHandler(e, { ...note, createdAt: getFormattedDate() })
```

Comment:
if yu can then put this function in a separate function not as a dynamic function in the jsx